araproc/framework/dataset.py

```diff
@@ -231,7 +248,28 @@ def __assign_config(self):
         self.qual_cuts = ROOT.AraQualCuts.Instance()
         self.config = self.qual_cuts.getLivetimeConfiguration(self.run_number, self.station_id)
 
-    def __load_pedestal(self):
+    def __load_pedestal(self, path_to_pedestal_file = None):
+
+        # find the right pedestal
+        # if the user provided one, then choose that
+        # otherwise, try to find it in cvmfs
+        # and if that doesn't work, raie an error
+        if path_to_pedestal_file is not None:
+            if file_is_safe(path_to_pedestal_file):
```

Comment:
Do we want a try-except here? I guess `file_is_safe` crashes the code if it isn't safe?

Reply:
I mean, that function has turned into something of a frankenstein... Let me see if I can clean it up a bit. It both returns a true/false, and raises, which isn't great practice really

Reply:
Okay, see what you think now?

Reply:
Looks good to me!